forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-mock-store/forgerock-openbanking-uk-aspsp-rs-mock-store-server/src/main/java/com/forgerock/openbanking/aspsp/rs/store/api/openbanking/account/v2_0/statements/StatementsApiController.java

```diff
@@ -183,29 +182,23 @@
                     OBRIErrorType.REQUEST_INVALID_HEADER.toOBError1("Invalid header 'Accept' the only supported value for this operation is '" + MediaType.APPLICATION_PDF_VALUE + "'"));
         }
 
-        // Check if this cusotmer has a statement file
+        // Get the statement file from GCP bucket
         Optional<Resource> statement = statementPDFService.getPdfStatement();
         if (statement.isPresent()) {
-            return ResponseEntity.ok()
-                    .contentLength(getContentLength(statement.get()))
-                    .contentType(MediaType.APPLICATION_PDF)
-                    .body(statement.get());
+            try {
+                return ResponseEntity.ok()
+                        .contentLength(statement.get().contentLength())
+                        .contentType(MediaType.APPLICATION_PDF)
+                        .body(statement.get());
+            } catch (IOException e) {
+                log.warn("We found a statement PDF file '{}' for ASPSP but could no get content-length with error", statement.get().getDescription(), e);
+                return null;
```

Comment:
I'm not sure if this is best response for a client? What about a server error?

Reply:
It was there, but I will improve the response in that case and I will add a new integration tests for that case.